cogs/birthday/birthday.py

```diff
@@ -510,56 +541,98 @@ async def setSelfBirthday(self, ctx: Context, *, birthday: MonthDayConverter):
             return
 
         birthdayStr = "{0:%B} {0:%d}".format(birthday)
-
         if birthdayConfig:
-            confirmationStr = (
-                f"Are you sure you want to set your birthday to "
-                f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}? "
-                "Only administrators and moderators can reset your birthday afterwards. "
-                f"Type {bold('`yes`', escape_formatting=False)} to confirm."
-            )
 
-            await ctx.author.send(f"{headerWarn}: {confirmationStr}")
+            async def mainFlow(channel: discord.TextChannel, carefree: bool = False):
+                confirmationStr = "\n".join(
+                    (
+                        f"Are you sure you want to set your birthday to "
+                        f"{spoiler(bold(birthdayStr, escape_formatting=False), escape_formatting=False)}? "
+                        "Only administrators and moderators can reset your birthday afterwards.",
+                        f"Type {bold('`yes`', escape_formatting=False)} to confirm.",
+                    )
+                )
+                # define the time for sensitive messages to live for
+                SENSITIVE_MSG_TTL = None if carefree else 5.0
+                try:
+                    await channel.send(
+                        f"{headerWarn}: {confirmationStr}",
+                        delete_after=SENSITIVE_MSG_TTL,
+                    )
+                except discord.Forbidden:
+                    # this means messages cannot be sent to the the destination channel
+                    # so we abort the flow now by re-raising the exception
+                    # and the caller shall handle it gracefully
+                    raise
 
-            def check(msg: discord.Message):
-                return msg.author == ctx.author and msg.channel == ctx.author.dm_channel
+                # wait for answer
+                def check(msg: discord.Message):
+                    return msg.author == ctx.author and msg.channel == channel
 
-            try:
-                response = await self.bot.wait_for("message", timeout=30.0, check=check)
-            except asyncio.TimeoutError:
-                # hide the birthday portion within the previous message
-                await ctx.author.send(
-                    f"{headerBad}: You took too long. Not setting your birthday."
+                try:
+                    response = await self.bot.wait_for("message", timeout=6.0, check=check)
```

Comment:
Thoughts on having a longer timeout for this one? We aren't dealing with sensitive info here so might as well leave the user a good amount of time just so they can read the prompt message and type their response.
This is related to the duration used for the prompt which contains sensitive info, so we can do a quick `if`-`else` like

```python
                timeout = SENSITIVE_MSG_TTL or 30
                # meaning we go for an ample amount of 30 secs if SENSITIVE_MSG_TTL is not set
                # and if it is set, take that as the timeout
                # or something like
                timeout = SENSITIVE_MSG_TTL + 1 if SENSITIVE_MSG_TTL else 30
```
and then
```suggestion
                    response = await self.bot.wait_for("message", timeout=timeout, check=check)
```